### PR TITLE
Clean up tests & Makefile

### PR DESCRIPTION
**Pull request by seanhagen**
Getting the full suite of gometalinter checks running, because reasons -- mostly being able to check for everything Hound does locally, as well as a few other things I'm not sure Hound checks for.

### CODE REVIEW

**Comment by seanhagen** [comment]
So, why was this failing?

Basically, every time Travis runs a build, it's building the environment from scratch. What that means is that gometalinter is installed with all of the latest linters *as of that moment*. Which means it probably has linters we aren't running locally. Which means they'll probably fail.

The solution is to run gometalinter with a whitelist of allowed linters in Travis, and to update the makefile so deps always uses `go get -u` to install gometalinter.

This way, we get all the new fun linters to test out locally, but Travis is only running the "important" ones.

File: app/authentication.go [hunk 1]
```diff
@@ -24,7 +24,9 @@ import (
 
 func getCreateSessionByTokenRoute(db *db.Database) httprouter.Handle {
 	return func(w http.ResponseWriter, r *http.Request, params httprouter.Params) {
-		defer r.Body.Close()
+		defer func() {
+			_ = r.Body.Close()
+		}()
```
Comment (rydrman) [inline app/authentication.go]: why the change here?

File: app/authentication.go [hunk 2]
```diff
@@ -42,7 +44,7 @@ func getCreateSessionByTokenRoute(db *db.Database) httprouter.Handle {
 			if err == sql.ErrNoRows {
 				log.Criticalf(c, "no matching tokens in database")
 				w.WriteHeader(http.StatusForbidden)
-				io.WriteString(w, "Invalid token")
+				_, _ = io.WriteString(w, "Invalid token")
```
Comment (rydrman) [inline app/authentication.go]: are these linting errors?

Comment (seanhagen) [inline app/authentication.go]: Yeah, this and the previous are linting errors -- apparently it's best practice to always check return values, especially if they're of type `error`. Doing this silences them, although after PoC we can have all errors getting sent to Honeybadger ( or just getting logged to Logstash ).

Comment (seanhagen) [inline app/authentication.go]: We should probably also have a conversation about exactly what linters we want to be running -- gometalinter has a lot of options, but some of them may be more intense than we care about right now.

**Comment by rydrman** [comment]
I'm not sure I'm a big fan of having all of the _ return vars, just seems like it makes the code more complex   in terms of readability... what's the reasoning behind it?

**Pull request merged**